src/main/java/Duke.java

```diff
@@ -48,61 +48,71 @@ public static void main(String[] args) {
+        printWelcomeMessage();
+        Task[] commands = createList();
         boolean not_finished = true;
         while (not_finished == true) {
             String line = ask();
             if (line.equals("bye")) {
-                System.out.println("____________________________________________________________\n");
-                System.out.println("\t Bye. Hope to see you again soon!\n");
-                System.out.println("____________________________________________________________\n");
+                printByeMessage();
                 not_finished = false;
             } else if (line.equals("list")) {
-                displayCommandsList(commands_list);
+                displayCommandsList(commands);
             } else if (line.length() > 5 && line.substring(0, 4).equals("mark")) {
                 int index = Integer.parseInt(line.substring(5, 6));
-                commands_list[index].setDone(true);
+                commands[index].setDone(true);
                 printDashLine();
-                System.out.println("\t Nice! I've marked this task as done:");
-                System.out.println("\t" + index + "." + commands_list[index]);
+                printSpecificTask(index, commands, "Nice! I've marked this task as done:");
                 printDashLine();
             } else if (line.length() > 7 && line.substring(0, 6).equals("unmark")) {
                 int index = Integer.parseInt(line.substring(7, 8));
-                commands_list[index].setDone(false);
+                commands[index].setDone(false);
                 printDashLine();
-                System.out.println("\t OK, I've marked this task as not done yet:");
-                System.out.println("\t" + index + "." + commands_list[index]);
+                printSpecificTask(index, commands, "OK, I've marked this task as not done yet:");
                 printDashLine();
             } else if (line.length() > 4 && line.substring(0, 4).equals("todo")) {
-                commands_list[next_index] = new Todo(line);
-                next_index++;
+                commands[nextIndexInList] = new Todo(line);
+                nextIndexInList++;
                 printDashLine();
-                System.out.println("\t Got it. I've added this task:");
-                System.out.println("\t" + commands_list[next_index - 1]);
+                printSpecificTask(nextIndexInList - 1, commands, "Got it. I've added this task:");
                 printDashLine();
             } else if (line.length() > 5 && line.substring(0, 5).equals("event")) {
-                int index_from = line.indexOf("/");
-                int index_to = line.indexOf("/", index_from + 1);
-                commands_list[next_index] = new Event(line.substring(0, index_from - 1),
-                        line.substring(index_from + 6, index_to - 1), line.substring(index_to + 4));
-                next_index++;
+                int indexFrom = line.indexOf("/");
```

Comment:
perhaps there could be other ways to do this that are more readable and less susceptible to mistakes. For example, using the <string instance>.split function.